sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -903,21 +903,46 @@ def hpat_pandas_series_shift(self, periods=1, freq=None, axis=0, fill_value=None
     if not isinstance(axis, (types.Omitted, int, types.Integer)):
         raise TypingError('{} Unsupported parameters. Given axis: {}'.format(_func_name, axis))
 
+    fill_is_default = isinstance(fill_value, (types.Omitted, types.NoneType)) or fill_value is None
+    series_np_dtype = [numpy_support.as_dtype(self.data.dtype)]
+    fill_np_dtype = [numpy.float64 if fill_is_default else numpy_support.as_dtype(fill_value)]
+
+    common_dtype = numpy.find_common_type(series_np_dtype, fill_np_dtype)
+
+    if fill_is_default:
+        def hpat_pandas_series_shift_impl(self, periods=1, freq=None, axis=0, fill_value=None):
+            if axis != 0:
+                raise TypingError('Method shift(). Unsupported parameters. Given axis != 0')
```

Comment:
I think TypingError should be changed to ValueError (or whatever runtime exception Pandas raises when it gets wrong axis). I would use TypingError during Typing only.

Reply:
TypingError doesn't rise and cannot be caught in tests for unknown reason